### PR TITLE
feat(di): Add the `@Injectable` annotation

### PR DESCRIPTION
**Pull request by kegluneq**
- Add an annotation marking a class as available to `Injector`s.
- Mark `Compiler` and its dependencies as available to the `Injector`.
- Modify hello_world to use `@Injectable`.

### CODE REVIEW

**Comment by pkozlowski-opensource** [comment]
@kegluneq hmm, sorry for a lame question, but why do we have a special marking of classes that can be part of the injector?

**Comment by kegluneq** [comment]
I don't know the answer, but it simplifies the dart/transform work I've been doing and in discussions with @mhevery yesterday he said it would be necessary on the Js side of things.

@mhevery , would you mind quickly outlining the necessity of the interface (or pointing to a doc, etc)?

**Comment by mhevery** [comment]
@pkozlowski-opensource we need it because TS will only annotate type information if decorator is present.

File: modules/angular2/src/di/annotations.js [hunk 1]
```diff
@@ -107,4 +107,24 @@ export class DependencyAnnotation {
...
\ No newline at end of file
+}
+
+/**
+ * A class annotation that marks a class as available to `Injector`s for
+ * creation.
+ *
+ * ```
+ * class NeedsService {
+ *   constructor(@Inject() svc:UsefulService) {}
```
Comment (mhevery) [inline modules/angular2/src/di/annotations.js]: This should just be  `constructor(svc:UsefulService)` no need for `@Inject()`

Comment (vicb) [inline modules/angular2/src/di/annotations.js]: But then would we have type infos ?

**Pull request closed without merging**